cmd/manager/scap.go

```diff
@@ -311,6 +323,34 @@ func fetch(client *kubernetes.Clientset, objects []utils.ResourcePath) (map[stri
 	return results, warnings, nil
 }
 
+func filter(ctx context.Context, rawobj []byte, filter string) ([]byte, error) {
+	fltr, fltrErr := gojq.Parse(filter)
+	if fltrErr != nil {
+		return nil, fmt.Errorf("could not create filter '%s': %w", filter, fltrErr)
+	}
+	obj := map[string]interface{}{}
+	unmarshallErr := json.Unmarshal(rawobj, &obj)
+	if unmarshallErr != nil {
+		return nil, fmt.Errorf("Error unmarshalling json: %w", unmarshallErr)
+	}
+	iter := fltr.RunWithContext(ctx, obj)
+	v, ok := iter.Next()
+	if !ok {
+		DBG("No result from filter, returning original object.")
```

Comment:
This debug statement is confusing..btw I agree with not returning anything, just an error, it is important to fail hard when filtering fails.

Reply:
aha! The original idea was to not fail here. Indeed this debug message no longer applies. Let me fix it.